### DISCUSSION TITLE
Binding to a text not working when I try to change the text from command?

### DISCUSSION THREAD

**Discussion by myplaths** [AvaloniaUI/Avalonia#7847] [Q&A]
I have this xaml code 
` <StackPanel>
        <TextBlock Text="{Binding MyText, Mode=TwoWay}"></TextBlock>
        <Button Command="{Binding SwitchViewCommand}" Content="Click me"></Button>
    </StackPanel>`

then my class inherits from viewmodelbase
`public ReactiveCommand<Unit, Unit> SwitchViewCommand { get; }

    
    public TestViewModel()
    {
        MyText = "default";
        SwitchViewCommand = ReactiveCommand.Create(SwitchViewAndUpdateDisplayText);
    }

    private string _myText;
    
    public string MyText
    {
        get => _myText;
        set => this.RaiseAndSetIfChanged(ref _myText, value);
    }

    private void SwitchViewAndUpdateDisplayText()
    {    
        MyText = "Is changed";
    }`
    
    How can I fix that? 
    
    When I press the button i want the text to be changed.

**Comment by maxkatz6** [comment]
Your code should work. Can you provide more complete sample?
Is SwitchViewAndUpdateDisplayText executed if you check it with debugger?

**Reply by myplaths** [reply]
could you maybe make a small project for me for this please? ive been smashing my head yesterday and i cannot figure out.

And do you know where can i find some avalonia projects so i can have more ideas.

**Reply by MarkusKgit** [reply]
There are some great articles from Nick Polyak you could dive into:
[Multiplatform Avalonia .NET Framework Programming Basic Concepts in Easy Samples](https://www.codeproject.com/Articles/5311995/Multiplatform-Avalonia-NET-Framework-Programming-B)
[Multiplatform Avalonia .NET Framework Programming Advanced Concepts in Easy Samples](https://www.codeproject.com/Articles/5317055/Multiplatform-Avalonia-NET-Framework-Programming-A)
[Basics of XAML in Easy Samples for Multiplatform Avalonia .NET Framework](https://www.codeproject.com/Articles/5314369/Basics-of-XAML-in-Easy-Samples-for-Multiplatform-A)

Then of course the [samples from Avalonia docs](https://docs.avaloniaui.net/tutorials/todo-list-app)

[More about ReactiveUI](https://www.reactiveui.net/docs/getting-started/) (which you don't have to use btw)

And last but not least [Awesome Avalonia](https://github.com/AvaloniaCommunity/awesome-avalonia)
